### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by adrianz-kainos**

### CODE REVIEW

File: test/handlers/api/softwareVersionsApiTest.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        });
+
+        it("should call softwareVersions dao with proper data", function() {
+            apiVersions.deleteVersions(req, res);
+
+            sinon.assert.calledOnce(versionsDao.delete);
+            expect(deleteVersionsSpy.getCalls()[0].args[0]).that.is.an('array').to.equal(testData.versions);
+        });
+    });
+});
```
Comment (xjedam) [inline test/handlers/api/softwareVersionsApiTest.js:172]: Might be worth adding some tests whether the data returned from dao gets properly translated to resultHandler or res.end+res.writeHead

File: test/handlers/web/technologiesTest.js [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+
+        it("should NOT render edit technology page if no technology was found", function() {
+            technology.getById.restore(); // get rid of the default stub created in beforeEach(){...}
+            sinon.stub(technology, 'getById', function(userId, num, cb){
+                cb(null /* missing technology object */);
+            });
+             
+            webTechnologies.edit(req, res);
+
+            sinon.assert.notCalled(res.render);
```
Comment (xjedam) [inline test/handlers/web/technologiesTest.js:85]: might be worth validating that the user is getting redirected to the error page here and in other examples.

File: test/handlers/web/technologiesTest.js [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+        });
+
+        it("should pass the technology object to the rendered view", function() {
+            webTechnologies.getVersions(req, res);
+
+            expect(res.render.args[0][1]).to.contain({'technology': technologyFromDb});
+        });
+
```
Comment (xjedam) [inline test/handlers/web/technologiesTest.js:171]: might be worth adding technology not found test

File: test/handlers/web/technologiesTest.js [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+            sinon.assert.notCalled(res.render);
+        });
+
+        it("should pass the technology object to the rendered view", function() {
+            webTechnologies.getTechnology(req, res);
+
+            sinon.assert.calledOnce(res.render);
+            expect(res.render.args[0][1]).to.contain({'technology': technologyFromDb});
+        });
```
Comment (xjedam) [inline test/handlers/web/technologiesTest.js:217]: might be worth adding technology not found test

File: test/handlers/web/technologiesTest.js [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+
+            sinon.assert.notCalled(res.render);
+        });
+        
+        it("should pass the technology object to the rendered view", function() {
+            webTechnologies.getUsers(req, res);
+
+            sinon.assert.calledOnce(res.render);
+            expect(res.render.args[0][1]).to.contain({'technology': technologyFromDb});
+        });
```
Comment (xjedam) [inline test/handlers/web/technologiesTest.js:286]: might be worth adding a test or two with a verification of user list for technology being fetched and passed to the renderer

File: test/handlers/web/technologiesTest.js [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+
+            sinon.assert.notCalled(res.render);
+        });
+
+        it("should pass the technology object to the rendered view", function() {
+            webTechnologies.getStatusHistory(req, res);
+
+            sinon.assert.calledOnce(res.render);
+            expect(res.render.args[0][1]).to.contain({'technology': technologyFromDb});
+        });
```
Comment (xjedam) [inline test/handlers/web/technologiesTest.js:345]: maybe some verification on what happens when tech not found - here,  and in some tests below

File: test/handlers/web/technologiesTest.js [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+            sinon.assert.notCalled(res.render);
+        });
+
+        it("should pass the technology object to the rendered view", function() {
+            webTechnologies.addProject(req, res);
+
+            sinon.assert.calledOnce(res.render);
+            expect(res.render.args[0][1]).to.contain({'technology': technologyFromDb});
+        });
+    });
```
Comment (xjedam) [inline test/handlers/web/technologiesTest.js:508]: perhaps we could add some tests for project.getAllForTechnology and project.getAll being called and outputs passed to the renderer

File: test/handlers/api/softwareVersionsApiTest.js (hunk 1 above)
Comment (adrianz-kainos) [reply test/handlers/api/softwareVersionsApiTest.js:172]: Done.

**Pull request merged**